virtme_ng/run.py

```diff
@@ -1084,6 +1093,25 @@ def _get_virtme_numa(self, args):
         else:
             self.virtme_param["numa"] = ""
 
+    def _get_virtme_numa_distance(self, args):
+        if args.numa_distance is not None:
+            if not args.numa:
+                arg_fail("error: --numa-distance can be used only with --numa", show_usage=False)
+            numa_dist_str = ""
+            for arg in args.numa_distance:
+                try:
+                    nodes = arg.split('=')
+                    src, dst = nodes[0].split(',')
+                    val = nodes[1]
+                    numa_dist_str += f" --numa-distance src={src},dst={dst},val={val}"
+                except ValueError:
+                    err_msg = f"error: invalid distance '{arg}', " + \
+                               "NUMA distance string must be in the format SRC,DST=VAL"
+                    arg_fail(err_msg, show_usage=False)
+            self.virtme_param["numa_distance"] = numa_dist_str
```

Comment:
Out of curiosity, why not doing the parsing and moving this "complexity" to `virtme-run`?

Reply:
> Out of curiosity, why not doing the parsing and moving this "complexity" to `virtme-run`?

At some point in the future I'd be nice to just drop virtme-run and have only vng. The virtme-run interface is maintained only for backward compatibility with the old virtme tool and vng is like a front-end to the old interface. But ultimately it'd be nice to drop the old interface and just have a front-end to qemu. That's the reason why I usually prefer to add complexity and new abstractions to vng directly.

Reply:
> > Out of curiosity, why not doing the parsing and moving this "complexity" to `virtme-run`?
> 
> At some point in the future I'd be nice to just drop virtme-run and have only vng. The virtme-run interface is maintained only for backward compatibility with the old virtme tool and vng is like a front-end to the old interface. But ultimately it'd be nice to drop the old interface and just have a front-end to qemu. That's the reason why I usually prefer to add complexity and new abstractions to vng directly.

Thank you, I see! And that's not what I did in my previous PR :)

Please note that I'm still using `virtme-run` directly in my automated scripts, mainly because it works, and I don't need `vng` to build the kernel (I already had that, and when I switched, `vng` could not build the kernel in a different build dir :) ).